contracts/voting-verifier/src/events.rs

```diff
@@ -177,6 +181,10 @@ fn parse_message_id(
     ))
 }
 
+pub fn construct_message_id(tx_hash: String, event_index: u32) -> String {
```

Comment:
this should accept `Hash` instead of `String`, since you are doing the conversion for the index in here anyway